Forms and input components updates (#27999)
dotnet/AspNetCore.Docs: aspnetcore/blazor/forms-and-input-components.md

* <xref:Microsoft.AspNetCore.Components.Forms.EditForm> component bound to a model that uses [data annotations](xref:mvc/models/validation)
* [Built-in input components](#built-in-input-components)

> The <xref:Microsoft.AspNetCore.Components.Forms?displayProperty=fullName> namespace provides classes for managing form elements, state, and validation, and the namespace also provides access to built-in :::no-loc text="Input*"::: Razor components for use in Blazor apps. The namespace appears by default in the `_Imports.razor` file of an app created from a Blazor project template. If you don't maintain a `using` statement in `_Imports.razor` for the namespace, you can use the namespace in individual Razor components with the [`@using`](xref:mvc/views/razor#using) directive:

Reviewer: ```suggestion
The <xref:Microsoft.AspNetCore.Components.Forms?displayProperty=fullName> namespace provides classes for managing form elements, state, and validation, as well as access to built-in :::no-loc text="Input*"::: components, which can be used in Blazor apps. Projects created from the Blazor project template include this namespace by default in the `_Imports.razor` file, which makes the namespace available in all the `.razor` files of the application without an explicit [`@using`](xref:mvc/views/razor#using) directive:
```

Author: I'm going to try again with this. The style manual doesn't favor "as well as" over "and," *and* I think we could split that into a list of two items. The shorter sentences and phrases machine translate better (so they tell us 😄).

```razor
@using Microsoft.AspNetCore.Components.Forms
```

:::moniker range=">= aspnetcore-7.0"

> To demonstrate how an <xref:Microsoft.AspNetCore.Components.Forms.EditForm> component works, consider the following `ExampleModel` type. `ExampleModel` defines a `Name` property for a form that accepts a name from the user.

Reviewer: ```suggestion
To demonstrate how an <xref:Microsoft.AspNetCore.Components.Forms.EditForm> component works, consider the following example. Here, the `ExampleModel` represents the data model the form will be bound to and defines a `Name` property which will be used to store the value of the `name` field of the form, provided by the user.
```

Author: Let's look at this revision, too. I'm trying to steer clear of the state verbs.

<EditForm Model="@exampleModel" OnSubmit="@HandleSubmit">
    <InputText id="name" @bind-Value="exampleModel.Name" />

    <button type="submit">Submit</button>
</EditForm>

@code {
    private ExampleModel exampleModel = new();

    private void HandleSubmit()
    {
        Logger.LogInformation("HandleSubmit called");

        // Process the form
    }
}
```

In the preceding `FormExample1` component:

* The <xref:Microsoft.AspNetCore.Components.Forms.EditForm> component is rendered where the `<EditForm>` element appears.
* The model is created in the component's `@code` block and held in a private field (`exampleModel`). The field is assigned to  <xref:Microsoft.AspNetCore.Components.Forms.EditForm.Model?displayProperty=nameWithType>'s attribute (`Model`) of the `<EditForm>` element.
* The <xref:Microsoft.AspNetCore.Components.Forms.InputText> component (`id="name"`) is an input component for editing string values. The `@bind-Value` directive attribute binds the `exampleModel.Name` model property to the <xref:Microsoft.AspNetCore.Components.Forms.InputText> component's <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601.Value%2A> property.

> * The `HandleSubmit` method is assigned to <xref:Microsoft.AspNetCore.Components.Forms.EditForm.OnSubmit>. The handler is called when the form is submitted by the user.

Reviewer: ```suggestion
* The `HandleSubmit` method is registered as a handler for the <xref:Microsoft.AspNetCore.Components.Forms.EditForm.OnSubmit> callback. The handler is called when the form is submitted by the user.
```
